tools/cli/commands/rsync.js

```diff
@@ -50,6 +52,53 @@ function setRsyncDest( pluginDestPath, alias = false ) {
  * @param {object} argv - The argv for the command line.
  */
 export async function rsyncInit( argv ) {
+	// Apple ships a special fork of openrsync, which has various quirks.
+	// In particular, it doesn't handle symlink recursion well, which breaks
+	// in macOS 15.4 and copies child node_modules dirs in 15.5.
+	//
+	// See also:
+	// * p1742486518169009-slack-CDLH4C1UZ
+	// * https://github.com/apple-oss-distributions/rsync/tree/main/openrsync
+	if ( os.platform() === 'darwin' ) {
+		const { stdout: rsyncVersion } = await execa( 'rsync', [ '--version' ] );
+		isOpenrsync = rsyncVersion.indexOf( 'openrsync' ) >= 0;
+		if ( isOpenrsync ) {
+			const { stdout: macOS_version } = await execa( 'sw_vers', [ '--productVersion' ] );
+			if ( macOS_version === '15.4' ) {
+				console.error(
+					chalk.red(
+						'The implementation of rsync in macOS 15.4 is unable to properly sync symlinks.'
+					)
+				);
+				console.error( chalk.red( 'Please install standard rsync (e.g. `brew install rsync`).' ) );
+				process.exit( 1 );
+			} else {
+				console.error(
+					chalk.yellow(
+						'The implementation of rsync in macOS is unable to properly sync symlinks.'
+					)
+				);
+				console.error(
+					chalk.yellow( 'Installing standard rsync (e.g. `brew install rsync`) is recommended.' )
+				);
+				console.error();
+				await enquirer
+					.prompt( {
+						type: 'confirm',
+						name: 'ignoreUntracked',
```

Comment:
It makes no difference to functionality, but "ignoreUntracked" doesn't seem like a good name for the variable here.